File: python/LinkedList.py
class Node:
    def __init__(self,data):
        self.data = data
        self.next = None

class LinkedList:
    def __init__(self):
        self.head = None
        self.tail = None
        
    def AddNode(self,data):
        new_node = Node(data)
        if self.head == None:
            self.head = new_node
        
        if self.tail != None:
            self.tail.next = new_node
        
        self.tail = new_node
    
    def RemNode(self,index):
        prev = None
        node = self.head
        i = 0
       
        while ( node != None ) and ( i < index ):
            prev = node
            node = node.next
            i += 1
       
        if prev == None:
            self.head = node.next
        else:
            prev.next = node.next        
    
    def PrintList(self):
        node = self.head
        while node != None:
            print node.data
            node = node.next
            
            
#
List = LinkedList()
List.AddNode(1)
List.AddNode(2)
List.AddNode(3)
List.AddNode(4)
List.PrintList( )

List.RemNode( 2 )
        
List.PrintList( )
'''
1
2
3
4
1
2
4
'''

File: C/linkedlist.c
#include <stdio.h>
#include <stdlib.h>

//Let us create a data type called node_t
typedef struct node{
    int data;           //can be any datatype, we just want to store some value
    struct node *ptr;   //Pointer to another node. Will be NULL for nothing to point
}node_t;

node_t *head;      //head is the pointer to the first node in the linked list

//O(1) insertion, means adding anode at the beginning
void addNode(int x){
    node_t *temp = (node_t*)malloc(sizeof(node_t));
    temp->data = x;
    /*if(head==NULL){         //number of elements is 
        temp->ptr = NULL; 
        head=temp;          //point to this node
    }
    else{
        temp->ptr = head;
        head = temp;
    }*/
    //Simply:
    temp->ptr = head;       //whether head is NUll or empty
    head=temp;
    
  
    //free(temp);
    
   
}

void print(){
    node_t *temp = (node_t*)malloc(sizeof(node_t));
    temp = head;
    printf("The list is");
    while(temp!=NULL){
        
        printf(" %d",temp->data);
        temp=temp->ptr;
    }
   printf("\n");
   
      
}

void reverse(){
 
}

void delNode();

int main(){
    head = (node_t*)malloc(sizeof(node_t));
    head = NULL;
    printf("How many numbers?");
    int n,i,x;
    scanf("%d",&n);
    for(i=0;i<n;i++){
        scanf("%d",&x);
        addNode(x);
        print();
    }
    print();
    //free(temp);
    return 0;
}

/*
How many numbers?4
23
The list is 23
34
The list is 34 23
56
The list is 56 34 23
109
The list is 109 56 34 23*/